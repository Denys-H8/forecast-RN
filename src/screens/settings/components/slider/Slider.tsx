import React, { useState, useEffect } from 'react';
import { View } from 'react-native';
import Slider from '@react-native-community/slider';
import Icon from 'react-native-vector-icons/FontAwesome';

import constants from 'src/constants';
import { CustomText } from 'src/components';
import { styles } from './slider.styles';

interface IProps {
  title: string;
  text: string;
  maximumValue: number;
  minimumValue: number;
  value: number;
  onChange(value: number): void;
}

const SliderView: React.FC<IProps> = ({ title, text, maximumValue, minimumValue, value, onChange }) => {
  const [icon, setIcon] = useState();

  useEffect(() => {
    Icon.getImageSource('circle', 16, 'white').then(setIcon);
  }, []);

  return (
    <View style={styles.container}>
      <View style={styles.subContainer}>
        <CustomText style={styles.text}>{title}</CustomText>
        <CustomText style={styles.subText}>{text}</CustomText>
      </View>

      <Slider
        testID="Slider.Slider"
        style={styles.slider}
        minimumValue={minimumValue}
        maximumValue={maximumValue}
        value={value}
        step={1}
        minimumTrackTintColor={constants.BTN_LOGIN_BG_COLOR}
        maximumTrackTintColor={constants.SUB_COLOR}
        thumbImage={icon}
        onValueChange={(value) => onChange(value)}
      />
    </View>
  );
};

export default SliderView;
