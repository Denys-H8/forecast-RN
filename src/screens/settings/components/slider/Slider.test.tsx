import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import Slider from './Slider';

const text = 'Some text value';
const title = 'New title value';
const maximumValue = 10;
const minimumValue = 0;
const value = 5;
const newValue = 7;

describe('InputField ', () => {
  test('should render InputField', () => {
    const onChangeMock = jest.fn();

    const { toJSON, getByTestId } = render(
      <Slider
        onChange={onChangeMock}
        text={text}
        title={title}
        maximumValue={maximumValue}
        minimumValue={minimumValue}
        value={value}
      />,
    );

    fireEvent(getByTestId('Slider.Slider'), 'onValueChange', newValue);

    expect(toJSON()).toMatchSnapshot();
    expect(onChangeMock).toHaveBeenCalled();
    expect(onChangeMock).toHaveBeenCalledWith(newValue);
    expect(onChangeMock).not.toHaveBeenCalledWith(9);
  });
});
