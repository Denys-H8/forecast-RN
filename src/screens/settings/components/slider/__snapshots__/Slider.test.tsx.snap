// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField  should render InputField 1`] = `
<View
  style={Object {}}
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "WorkSans-Regular",
          },
          Object {
            "color": "#fff",
            "fontFamily": "WorkSans-SemiBold",
            "fontSize": 22,
            "lineHeight": 36,
          },
        ]
      }
    >
      New title value
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "WorkSans-Regular",
          },
          Object {
            "color": "#fff",
            "fontSize": 22,
            "lineHeight": 36,
          },
        ]
      }
    >
      Some text value
    </Text>
  </View>
  <RNCSlider
    disabled={false}
    enabled={true}
    inverted={false}
    maximumTrackTintColor="#8B8EA5"
    maximumValue={10}
    minimumTrackTintColor="#7B61FF"
    minimumValue={0}
    onChange={[Function]}
    onRNCSliderSlidingComplete={null}
    onRNCSliderSlidingStart={null}
    onRNCSliderValueChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    step={1}
    style={
      Array [
        Object {
          "height": 40,
        },
        Object {
          "height": 40,
        },
      ]
    }
    testID="Slider.Slider"
    thumbImage={null}
    value={5}
  />
</View>
`;
